asyncpg/protocol/protocol.pyx

```diff
@@ -393,6 +434,9 @@ cdef class BaseProtocol(CoreProtocol):
                 'cannot perform operation: another operation is in progress')
 
     cdef _new_waiter(self, timeout):
+        if self.waiter is not None:
```

Comment:
This is a piece that I added during debugging this PR. I think we should have it, but it should go in a separate commit.